File: src/main/java/io/github/alexiscomete/lapinousecond/view/AnswerManager.java
package io.github.alexiscomete.lapinousecond.view;

import org.json.JSONObject;
import java.io.InputStream;
import java.nio.charset.StandardCharsets;
import java.util.Scanner;

public class AnswerManager {
    private final JSONObject jsonObject;

    public AnswerManager(InputStream input) {
        if (input != null) {
            Scanner sc = new Scanner(input, StandardCharsets.UTF_8);
            StringBuilder builder = new StringBuilder();
            sc.forEachRemaining((s) -> builder.append(s).append(" "));
            System.out.println(builder);
            jsonObject = new JSONObject(builder.toString());
            sc.close();
        } else {
            jsonObject = new JSONObject("{}");
        }
    }

    public String getAnswer(LangageEnum langageEnum, AnswerEnum answerEnum) {
        JSONObject object = jsonObject.getJSONObject(answerEnum.getName());
        if (object == null) {
            return langageEnum.getInvalidAnswer();
        }
        String answer = object.getString(langageEnum.getName());
        if (answer == null || answer.equals("")) {
            return langageEnum.getInvalidAnswer();
        }
        System.out.println(answer);
        return answer;
    }

    public String formatAnswer(String answer, Object... format) {
        int i = 1;
        for (Object form :
                format) {
            answer = answer.replace("replace" + i, form.toString());
            i++;
        }
        return answer;
    }
}

File: src/main/java/io/github/alexiscomete/lapinousecond/view/AnswerManager.kt
package io.github.alexiscomete.lapinousecond.view

import org.json.JSONObject
import java.io.InputStream
import java.util.*

class AnswerManager(input: InputStream?) {
    private var jsonObject: JSONObject? = null

    init {
        if (input != null) {
            val sc = Scanner(input)
            val builder = StringBuilder()
            sc.forEachRemaining { s: String? -> builder.append(s).append(" ") }
            println(builder)
            jsonObject = JSONObject(builder.toString())
            sc.close()
        } else {
            jsonObject = JSONObject("{}")
        }
    }

    fun getAnswer(langageEnum: LangageEnum, answerEnum: AnswerEnum): String {
        val `object` = jsonObject!!.getJSONObject(answerEnum.value) ?: return langageEnum.invalidAnswer
        val answer = `object`.getString(langageEnum.displayName)
        if (answer == null || answer == "") {
            return langageEnum.invalidAnswer
        }
        println(answer)
        return answer
    }

    fun formatAnswer(answer: String, format: ArrayList<String>): String {
        var answer1 = answer
        var i = 1
        for (form in format) {
            println(form.javaClass.name)
            answer1 = answer1.replace("replace$i", form)
            i++
        }
        return answer1
    }
}